src/main/java/io/cdap/plugin/gcp/gcs/sink/GCSBatchSink.java

```diff
@@ -137,12 +144,25 @@ private void emitMetrics(boolean succeeded, BatchSinkContext context) {
 
     try {
       StorageClient storageClient = StorageClient.create(config.getProject(), config.getServiceAccountFilePath());
-      storageClient.mapMetaDataForAllBlobs(outputPath, new MetricsEmitter(context.getMetrics())::emitMetrics);
+      storageClient.mapMetaDataForAllBlobs(getPrefixPath(), new MetricsEmitter(context.getMetrics())::emitMetrics);
     } catch (Exception e) {
       LOG.warn("Metrics for the number of affected rows in GCS Sink maybe incorrect.", e);
     }
   }
 
+  private String getPrefixPath() {
+    if (!config.getFileSystemProperties().containsKey(AVRO_NAMED_OUTPUT)) {
+      return outputPath;
+    }
+    //Following code is for handling a special case for saving files in same output directory with avro format.
+    //The specified file prefix from Filesystem Properties will be used to make the filename unique.
+    //The code is based on assumptions from the internal implementation of
+    //org.apache.avro.mapreduce.AvroOutputFormatBase and org.apache.hadoop.mapreduce.lib.output.FileOutputFormat
+    return String
+      .format("%s/%s-", outputPath.endsWith("/") ? outputPath.substring(0, outputPath.length() - 1) :
```

Comment:
nit: it's a little difficult to read, can you create variables for the 2 values after the ?

Reply:
Done